### PR TITLE
fix(newsletter): Reply count

### PR DESCRIPTION
**Pull request by IgboPharaoh**
Get post replies from combined summary

### CODE REVIEW

File: src/app/newsletter/page.tsx [hunk 1]
```diff
@@ -3,16 +3,37 @@ import { NewsLetterDataType, NewsLetterSet } from "@/helpers/types";
 import { NewsletterPage } from "../components/server/newsletter";
 import Link from "next/link";
 import { formattedDate } from "@/helpers/utils";
+import { PRODUCTION_URL } from '@/config/config';
+import { getSummaryData } from '../summary/[...path]/page';
 
-// get most recent newsletter from newsletter.json
-const getCurrentNewsletter = () => {
+/** get most recent newsletter from newsletter.json **/
```
Comment (Extheoisah) [inline src/app/newsletter/page.tsx:9]: nit: let's stick one comment style in the code
```suggestion
// get most recent newsletter from newsletter.json
```

File: src/app/newsletter/page.tsx [hunk 2]
```diff
@@ -3,16 +3,37 @@ import { NewsLetterDataType, NewsLetterSet } from "@/helpers/types";
...
+    const data = fs.readFileSync(`${process.cwd()}/public/static/static/newsletters/newsletter.json`, "utf-8");
     const parsedData = JSON.parse(data) as NewsLetterDataType;
-    return parsedData;
+    let postParsedData: any = {};
+
+    for (const [key, value] of Object.entries(parsedData)) {
+      if (Array.isArray(value)) {
+        const processedValue = await Promise.all(
+          value.map(async (post) => {
+            /**  get the file path of all posts in the array **/
```
Comment (Extheoisah) [inline src/app/newsletter/page.tsx:20]:
```suggestion
            // get the file path of all posts in the array
```

File: src/app/newsletter/page.tsx [hunk 3]
```diff
@@ -3,16 +3,37 @@ import { NewsLetterDataType, NewsLetterSet } from "@/helpers/types";
...
+            const filePath = post.contributors.length > 0 ? post.combined_summ_file_path : post.file_path;
+            const file = filePath.replace(PRODUCTION_URL, "");
+
+            // /**  read the content of the file in the file path  **/
```
Comment (Extheoisah) [inline src/app/newsletter/page.tsx:24]:
```suggestion
            // read the content of the file in the file path
```

**Review by Extheoisah — CHANGES_REQUESTED** [review]
Look good! Just a few nits

**Comment by IgboPharaoh** [comment]
> Look good! Just a few nits

Alright, will make the necessary changes

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**